src/Yoti/Http/AbstractRequestHandler.php

```diff
@@ -43,50 +47,106 @@ abstract class AbstractRequestHandler
-     * @param string $sdkIdentifier
+     * @param string $sdkIdentifier - deprecated - use ::setSdkIdentifier() instead.
      *
      * @throws RequestException
      */
-    public function __construct($connectApiUrl, $pem, $sdkId, $sdkIdentifier)
+    public function __construct($apiUrl, $pem, $sdkId = null, $sdkIdentifier = null)
     {
-        $this->pem = $pem;
+        $this->pemFile = PemFile::fromString($pem);
         $this->sdkId = $sdkId;
-        $this->connectApiUrl = $connectApiUrl;
-        $this->sdkIdentifier = $sdkIdentifier;
+        $this->apiUrl = rtrim($apiUrl, '/');
 
-        $this->authKey = $this->extractAuthKeyFromPemContent();
+        if (isset($sdkIdentifier)) {
+            $this->setSdkIdentifier($sdkIdentifier);
+        } else {
+            $this->sdkIdentifier = self::YOTI_SDK_IDENTIFIER;
+        }
     }
 
     /**
      * @param string $endpoint
      * @param string $httpMethod
      * @param Payload|NULL $payload
+     * @param array queryParams
      *
      * @return array
      *
      * @throws RequestException
      */
-    public function sendRequest($endpoint, $httpMethod, Payload $payload = null)
-    {
+    public function sendRequest(
+        $endpoint,
+        $httpMethod,
+        Payload $payload = null,
+        array $queryParams = []
+    ) {
         self::validateHttpMethod($httpMethod);
 
-        $signedDataArr = RequestSigner::signRequest($this, $endpoint, $httpMethod, $payload);
+        // Ensure endpoint always has a single leading slash.
+        $endpoint = '/' . ltrim($endpoint, '/');
+
+        $signedDataArr = RequestSigner::signRequest($this, $endpoint, $httpMethod, $payload, $queryParams);
         $requestHeaders = $this->generateRequestHeaders($signedDataArr[RequestSigner::SIGNED_MESSAGE_KEY]);
-        $requestUrl = $this->connectApiUrl . $signedDataArr[RequestSigner::END_POINT_PATH_KEY];
+        $requestUrl = $this->apiUrl . $signedDataArr[RequestSigner::END_POINT_PATH_KEY];
 
         return $this->executeRequest($requestHeaders, $requestUrl, $httpMethod, $payload);
     }
 
     /**
-     * @return string
+     * Performs GET request.
+     *
+     * @param string $endpoint
+     * @param array queryParams
+     *
+     * @return array
+     *
+     * @throws RequestException
+     */
+    public function get($endpoint, array $queryParams = [])
+    {
+        return $this->sendRequest($endpoint, self::METHOD_GET, null, $queryParams);
+    }
+
+    /**
+     * Performs POST request.
+     *
+     * @param string $endpoint
+     * @param Payload|NULL $payload
+     * @param array queryParams
+     *
+     * @return array
+     *
+     * @throws RequestException
+     */
+    public function post($endpoint, Payload $payload = null, array $queryParams = [])
+    {
+        return $this->sendRequest($endpoint, self::METHOD_POST, $payload, $queryParams);
+    }
+
+    /**
+     * @deprecated will be removed in version 3 - SDK ID is now added as a query param.
```

Comment:
Would be good if we said _where_ it's added as a query param too, since it's not in this file 👍

Reply:
Fixed in 2a8784b / 1aacef5